src/main/Application.cpp

```diff
@@ -45,30 +45,58 @@ Application::create(VirtualClock& clock, Config const& cfg, bool newDB,
 }
 
 bool
-Application::modeHasOverlay(AppMode m)
+Application::modeUsesInMemoryDB(AppMode m)
 {
     switch (m)
     {
     case AppMode::RUN_LIVE_NODE:
+        return false;
+    case AppMode::RUN_CONSENSUS_ONLY:
+    case AppMode::REPLAY_IN_MEMORY:
         return true;
-    case AppMode::RELAY_LIVE_TRAFFIC:
-        return true;
+    default:
+        throw std::runtime_error("unhandled application mode");
+    }
+}
+
+bool
+Application::modeHasBucketList(AppMode m)
+{
+    switch (m)
+    {
+    case AppMode::RUN_LIVE_NODE:
     case AppMode::REPLAY_IN_MEMORY:
+        return true;
+    case AppMode::RUN_CONSENSUS_ONLY:
         return false;
     default:
         throw std::runtime_error("unhandled application mode");
     }
 }
 
 bool
-Application::modeHasDatabase(AppMode m)
+Application::modeStoresHistoricalData(AppMode m)
 {
     switch (m)
     {
     case AppMode::RUN_LIVE_NODE:
         return true;
-    case AppMode::RELAY_LIVE_TRAFFIC:
+    case AppMode::RUN_CONSENSUS_ONLY:
+    case AppMode::REPLAY_IN_MEMORY:
         return false;
+    default:
+        throw std::runtime_error("unhandled application mode");
+    }
+}
+
+bool
+Application::modeCommitsLedgerTxnRoot(AppMode m)
```

Comment:
The mode would be easier to understand if we just call it "enable in memory ledger"